lib/src/snapd_client.dart

```diff
@@ -63,1000 +64,289 @@ class SnapdException implements Exception {
-
-  @override
-  String toString() =>
-      'SnapPublisher(id: $id, username: $username, displayName: $displayName, validation: $validation)';
-
-  @override
-  bool operator ==(Object other) {
-    if (identical(this, other)) return true;
-
-    return other is SnapPublisher &&
-        other.id == id &&
-        other.username == username &&
-        other.displayName == displayName &&
-        other.validation == validation;
-  }
-
-  @override
-  int get hashCode => Object.hash(id, username, displayName, validation);
 }
 
 /// Describes a piece of media associated with a snap.
-@immutable
-@JsonSerializable()
-class SnapMedia {
-  const SnapMedia({
-    required this.type,
-    required this.url,
-    this.width,
-    this.height,
-  });
-
+@freezed
+class SnapMedia with _$SnapMedia {
+  const factory SnapMedia({
+    required String type,
+    required String url,
+    int? width,
+    int? height,
+  }) = _SnapMedia;
   factory SnapMedia.fromJson(Map<String, dynamic> json) =>
       _$SnapMediaFromJson(json);
-
-  /// Media type
-  final String type;
-
-  /// URL of media.
-  final String url;
-
-  /// Width of media in pixels.
-  final int? width;
-
-  /// Height of media in pixels.
-  final int? height;
-
-  Map<String, dynamic> toJson() => _$SnapMediaToJson(this);
-
-  @override
-  String toString() =>
-      'SnapMedia(type: $type, url: $url, width: $width, height: $height)';
-
-  @override
-  bool operator ==(Object other) {
-    if (identical(this, other)) return true;
-
-    return other is SnapMedia &&
-        other.type == type &&
-        other.url == url &&
-        other.width == width &&
-        other.height == height;
-  }
-
-  @override
-  int get hashCode => Object.hash(type, url, width, height);
 }
 
-/// Describes a snap package.
-@immutable
-@JsonSerializable()
-class Snap {
-  const Snap({
-    required this.name,
-    this.apps = const [],
-    this.base,
-    this.categories = const [],
-    this.channel = '',
-    this.channels = const {},
-    this.commonIds = const [],
-    this.confinement = SnapConfinement.unknown,
-    this.contact = '',
-    this.description = '',
-    this.devmode = false,
-    this.downloadSize,
-    this.hold,
-    this.id = '',
-    this.installDate,
-    this.installedSize,
-    this.jailmode = false,
-    this.license,
-    this.media = const [],
-    this.mountedFrom,
-    this.private = false,
-    this.publisher,
-    this.revision = '',
-    this.status = SnapStatus.unknown,
-    this.storeUrl,
-    this.summary = '',
-    this.title = '',
-    this.trackingChannel,
-    this.tracks = const [],
-    this.type = '',
-    this.version = '',
-    this.website,
-  });
+@freezed
+class Snap with _$Snap {
+  const factory Snap({
+    required String id,
+    required String name,
+    required String revision,
+    required String version,
+    required String channel,
+    required String contact,
+    required String description,
```

Comment:
Should those be optional as well?
It feels a bit inconsistent now with `String? title`, `@Default('') String summary` and `required String description`